streaming/src/test/scala/org/apache/spark/streaming/StreamingListenerSuite.scala

```diff
@@ -140,6 +141,16 @@ class StreamingListenerSuite extends TestSuiteBase with Matchers {
     }
     true
   }
+
+  test("registering listeners via spark.streaming.extraListeners") {
+    val conf = new SparkConf().setMaster("local").setAppName("test")
+      .set("spark.streaming.extraListeners", classOf[BatchInfoCollector].getName + "," +
```

Comment:
The change is already in StreamingListenerSuite.scala (and the class
StreamingListenerSuite) which seems appropriate. Did you mean create new
classes within StreamingListenerSuite.scala to handle the test cases?

On Mon, Jul 13, 2015 at 5:36 PM, Tathagata Das <notifications@github.com>
wrote:

> In
> streaming/src/test/scala/org/apache/spark/streaming/StreamingListenerSuite.scala
> <https://github.com/apache/spark/pull/6380#discussion_r34524259>:
>
> > @@ -140,6 +141,16 @@ class StreamingListenerSuite extends TestSuiteBase with Matchers {
> >      }
> >      true
> >    }
> > +
> > +  test("registering listeners via spark.streaming.extraListeners") {
> > +    val conf = new SparkConf().setMaster("local").setAppName("test")
> > +      .set("spark.streaming.extraListeners", classOf[BatchInfoCollector].getName + "," +
>
> I missed it earlier, this unit does not cover the two cases, one without
> constructor and one with constructor.
> It may be easier to read if you create new classes -
> StreamingListenerWithoutArg, and StreamingListenerWithConf
> (similar to ListenerThatAcceptsSparkConf for spark.extraListeners in
> SparkListenerSuite)
>
> Also, the StreamingContextSuite is getting really overloaded, mind
> shifting this to StreamingListenerSuite, and reverting this file?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/6380/files#r34524259>.
>